includes/class-amp-theme-support.php

```diff
@@ -1531,7 +1533,11 @@ public static function ensure_required_markup( Document $dom, $script_handles =
 		}
 
 		// When opting-in to POST forms, omit the amp-form component entirely since it blocks submission.
-		if ( amp_is_native_post_form_allowed() && $dom->xpath->query( '//form[ @action and @method and translate( @method, "POST", "post" ) = "post" ]' )->length > 0 ) {
+		if (
```

Comment:
Correct, because now that the `AMP_Script_Sanitizer` can modify the args of `AMP_Form_Sanitizer` to enable the `native_post_forms_allowed` arg, the return value of `amp_is_native_post_form_allowed()` may not be accurate at this point. It's return value is used as the initial value for `native_post_forms_allowed`, but since it can be changed by a sanitizer then it's only relevant for setting the initial value in `amp_get_content_sanitizers()`. 

In fact, there's not really a need for a global `amp_is_native_post_form_allowed()` function anymore.

The only reason why the function was here was to offer a slight performance improvement to skip doing the XPath query. But now we are checking to see if the `amp-form` extension was identified instead.